test/unit/controllers/declaration/DeclarationHolderRequiredControllerSpec.scala

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package unit.controllers.declaration
+
+import controllers.declaration.{routes, DeclarationHolderRequiredController}
+import forms.common.{Eori, YesNoAnswer}
+import forms.declaration.DeclarationHolder
+import models.DeclarationType.{CLEARANCE, OCCASIONAL, SIMPLIFIED, STANDARD, SUPPLEMENTARY}
+import models.Mode
+import org.mockito.ArgumentCaptor
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.{reset, times, verify, when}
+import org.scalatest.OptionValues
+import play.api.data.Form
+import play.api.mvc.{AnyContentAsEmpty, Request}
+import play.api.test.Helpers._
+import play.twirl.api.HtmlFormat
+import unit.base.ControllerSpec
+import views.html.declaration.declarationHolder.declaration_holder_required
+
+class DeclarationHolderRequiredControllerSpec extends ControllerSpec with OptionValues {
+
+  val mockPage = mock[declaration_holder_required]
+
+  val controller = new DeclarationHolderRequiredController(
+    mockAuthAction,
+    mockJourneyAction,
+    mockExportsCacheService,
+    navigator,
+    stubMessagesControllerComponents(),
+    mockPage
+  )
+
+  override protected def beforeEach(): Unit = {
+    super.beforeEach()
+    authorizedUser()
+    when(mockPage.apply(any(), any())(any(), any())).thenReturn(HtmlFormat.empty)
+  }
+
+  override protected def afterEach(): Unit = {
+    reset(mockPage)
+    super.afterEach()
+  }
+
+  override def getFormForDisplayRequest(request: Request[AnyContentAsEmpty.type]): Form[_] = {
+    withNewCaching(aDeclaration(withDeclarationHolders()))
+    await(controller.displayPage(Mode.Normal)(request))
+    theResponseForm
+  }
+
+  def theResponseForm: Form[YesNoAnswer] = {
+    val captor = ArgumentCaptor.forClass(classOf[Form[YesNoAnswer]])
+    verify(mockPage).apply(any(), captor.capture())(any(), any())
+    captor.getValue
+  }
+
+  private def verifyPageInvoked(numberOfTimes: Int = 1): HtmlFormat.Appendable =
+    verify(mockPage, times(numberOfTimes)).apply(any(), any())(any(), any())
+
+  "DeclarationHolder Required Controller" should {
+
+    onEveryDeclarationJourney() { request =>
+      "return 200 (OK)" that {
```

Comment:
Could tests for different endpoints be separated? In my opinion it is harder to read when it is not mentioned in the test name and you need to actually read the test's definition to know if it is about GET, or POST endpoints.
In this case it is about different methods, not endpoints, to be precise.

Reply:
Hope I got what you want here. In case not :blush: let's chat about it.